packages/accounts-base/accounts_server.js

```diff
@@ -1330,7 +1333,8 @@ const defaultResumeLoginHandler = (accounts, options) => {
         {"services.resume.loginTokens.hashedToken": hashedToken},
         {"services.resume.loginTokens.token": options.resume}
       ]
-    });
+    },
+    {fields: {"services.resume.loginTokens": 1}});
```

Comment:
Another nitpick, but this could even be more limited with a positional operator, thus:
`{fields: {"services.resume.loginTokens.$": 1}}`
Same for the one on line https://github.com/meteor/meteor/pull/10818/files#diff-2ae5b6c36ab4132c1a2d0e33b5fd8443R1323

Reply:
Yes, in the simple query, but doing it with the `$or` query fails the tests because
```
    user = accounts.users.findOne({
      $or: [
        {"services.resume.loginTokens.hashedToken": hashedToken},
        {"services.resume.loginTokens.token": options.resume}
      ]
    },
    {fields: {"services.resume.loginTokens.$": 1}});
```
returns `{ _id: ..., services: { resume: {} } }`

I think because of [this](https://docs.mongodb.com/manual/reference/operator/projection/positional/#array-field-limitations) limitation of the positional operator:

> The query document should only contain a single condition on the array field being projected. Multiple conditions may override each other internally and lead to undefined behaviour. 

Is there another way of filtering the returned array within mongo?  I'm playing with `$elemMatch` but no luck so far with nested fields and `$or`...

Reply:
Yep, you're right for the $or case and afaik, there's no other way to do this.

Reply:
Pushed a commit with the positional operator in the the simple case, and added a comment to avoid it in the `$or` case.